### ISSUE TITLE
Qualys Scan report import  fails to decode CVSS field

### ISSUE THREAD

**Issue by kareem-DA** [DefectDojo/django-DefectDojo#4319]
**Bug description**

I am using docker-compose to deploy.  I currently have the running on DDJ 1.8.0, but I have been testing on version 1.14.1 and It has the same issue.  

I am using the "Qualys Scan" import when I see this.  

I am using the API with a python script to to do the import.

so, I am   
       # Prior section, I get the report from the Qulays API store it into a byteIO object named "storage"
       # Get the project ID
        project_id = return_product_id(dd, dojo_project)
        # Get an engagement_id
        engagement_id = return_engagement(dd, project_id, args.dojoUser)

        active = True
        verified = True
        close_old_findings = True
        skip_duplicates = True
        scan_date = datetime.now().strftime("%Y-%m-%d")
        build = None
        scan_type = "Qualys Scan"
        minimum_severity = "High"
        push_to_jira = False
        tags = None

        data = {
            "file": storage.getvalue(),
            "engagement": ("", engagement_id.data["id"]),
            "scan_type": ("", scan_type),
            "active": ("", active),
            "verified": ("", verified),
            "close_old_findings": ("", close_old_findings),
            "skip_duplicates": ("", skip_duplicates),
            "scan_date": ("", scan_date),
            "tags": ("", tags),
            "build_id": ("", build),
            "minimum_severity": ("", minimum_severity),
            "push_to_jira": ("", push_to_jira),
        }
        response = dd._request("POST", "import-scan/", files=data)
        if response.success is False:
            _logger.error(
                "Posted Qualys results to DDJ failed. Response message: {}".format(
                    response.message
                )
            )
        dd.close_engagement(engagement_id.data["id"])

When I import a qualys scan, I am getting an error.  

```log
uwsgi_1         | ERROR:django.request:Internal Server Error: /api/v2/import-scan/
uwsgi_1         | Traceback (most recent call last):
uwsgi_1         |   File "./dojo/api_v2/serializers.py", line 1097, in save
uwsgi_1         |     parser_findings = parser.get_findings(data.get('file', None), test)
uwsgi_1         |   File "./dojo/tools/qualys/parser.py", line 233, in get_findings
uwsgi_1         |     return qualys_parser(file)
uwsgi_1         |   File "./dojo/tools/qualys/parser.py", line 216, in qualys_parser
uwsgi_1         |     master_list += issue_r(issue, d)
uwsgi_1         |   File "./dojo/tools/qualys/parser.py", line 154, in issue_r
uwsgi_1         |     if _temp['CVSS_score'] is not None and float(_temp['CVSS_score']) > 0:
uwsgi_1         | ValueError: could not convert string to float: '4.3 (AV:N/AC:M/Au:N/C:P/I:N/A:N)'
uwsgi_1         | 
```

Looking at the report. I see records where the CVSS field isn't a float.   

```XML
<VULN_DETAILS id="qid_11827">
        <QID id="qid_11827">11827</QID>
        <TITLE><![CDATA[HTTP Security Header Not Detected]]></TITLE>
        <SEVERITY>2</SEVERITY>
        <CATEGORY>CGI</CATEGORY>
        <THREAT><![CDATA[This QID reports the absence of the following <A HREF="https://www.owasp.org/index.php/OWASP_Secure_Headers_Project#tab=Headers" TARGET="_blank">HTTP headers</A> according to <A HREF="https://cwe.mitre.org/data/definitions/693.html" TARGET="_blank">CWE-693: Protection Mechanism Failure</A>:<BR>
X-Frame-Options: This HTTP response header improves the protection of web applications against clickjacking attacks. Clickjacking, also known as a &quot;UI redress attack&quot;, allows an attacker to use multiple transparent or opaque layers to trick a targeted user into clicking on a button or link on another page when they were intending to click on the top level page. <BR>
X-XSS-Protection: This HTTP header enables the browser built-in Cross-Site Scripting (XSS) filter to prevent cross-site scripting attacks. X-XSS-Protection: 0; disables this functionality.<BR>
X-Content-Type-Options: This HTTP header prevents attacks based on MIME-type mismatch. The only possible value is nosniff. If your server returns X-Content-Type-Options: nosniff in the response, the browser will refuse to load the styles and scripts in case they have an incorrect MIME-type. <BR>
Strict-Transport-Security: The HTTP Strict-Transport-Security response header (HSTS) is a security feature that lets a website tell browsers that it should only be communicated with using HTTPS, instead of using HTTP.<P>
QID Detection Logic:<BR>
This unauthenticated QID looks for the presence of the following HTTP responses:<BR>
Valid directives for X-Frame-Options are:<BR>
X-Frame-Options: DENY - The page cannot be displayed in a frame, regardless of the site attempting to do so.<BR>
X-Frame-Options: SAMEORIGIN - The page can only be displayed in a frame on the same origin as the page itself.<BR>
X-Frame-Options: ALLOW-FROM RESOURCE-URL - The page can only be displayed in a frame on the specified origin.<P>
Valid directives for X-XSS-Protections are:<BR>
X-XSS-Protection: 1 - Enables XSS filtering (usually default in browsers). If a cross-site scripting attack is detected, the browser will sanitize the page (remove the unsafe parts).<BR>
X-XSS-Protection: 1; mode=block - Enables XSS filtering. Rather than sanitizing the page, the browser will prevent rendering of the page if an attack is detected.<BR>
X-XSS-Protection: 1; report=URI  - Enables XSS filtering. If a cross-site scripting attack is detected, the browser will sanitize the page and report the violation. This uses the functionality of the CSP report-uri directive to send a report.<BR>
X-XSS-Protection: 0 disables this directive and hence is also treated as not detected.<P>
A valid directive for X-Content-Type-Options: nosniff<P>
A valid HSTS directive Strict-Transport-Security: max-age=&lt;expire-time&gt;; [; includeSubDomains][; preload]<P>
<B>NOTE:</B> All report-only directives (where applicable) are considered invalid.]]></THREAT>
        <IMPACT><![CDATA[Depending on the vulnerability being exploited, an unauthenticated remote attacker could conduct cross-site scripting, clickjacking or MIME-type sniffing attacks.]]></IMPACT>
        <SOLUTION><![CDATA[<B>Note:</B> To better debug the results of this QID, it is requested that customers execute commands to simulate the following functionality: curl -lkL --verbose.<P>
CWE-693: Protection Mechanism Failure mentions the following - The product does not use or incorrectly uses a protection mechanism that provides sufficient defense against directed attacks against the product. A &quot;missing&quot; protection mechanism occurs when the application does not define any mechanism against a certain class of attack. An &quot;insufficient&quot; protection mechanism might provide some defenses - for example, against the most common attacks - but it does not protect against everything that is intended. Finally, an &quot;ignored&quot; mechanism occurs when a mechanism is available and in active use within the product, but the developer has not applied it in some code path.<P>
Customers are advised to set proper <A HREF="https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/X-Frame-Options" TARGET="_blank">X-Frame-Options</A>, <A HREF="https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/X-XSS-Protection" TARGET="_blank">X-XSS-Protection</A>, <A HREF="https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/X-Content-Type-Options" TARGET="_blank">X-Content-Type-Options</A> and <A HREF="https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Strict-Transport-Security" TARGET="_blank">Strict-Transport-Security</A> HTTP response headers.<P>
Depending on their server software, customers can set directives in their site configuration or Web.config files. Few examples are:<P>
X-Frame-Options:<BR>
Apache: Header always append X-Frame-Options SAMEORIGIN<BR>
nginx: add_header X-Frame-Options SAMEORIGIN;<BR>
HAProxy: rspadd X-Frame-Options:\ SAMEORIGIN<BR>
IIS: &lt;HTTPPROTOCOL&gt;&lt;CUSTOMHEADERS&gt;&lt;ADD NAME=&quot;X-Frame-Options&quot; VALUE=&quot;SAMEORIGIN&quot;&gt;&lt;/ADD&gt;&lt;/CUSTOMHEADERS&gt;&lt;/HTTPPROTOCOL&gt;<P>
X-XSS-Protection:<BR>
Apache: Header always set X-XSS-Protection &quot;1; mode=block&quot; <BR>
PHP: header(&quot;X-XSS-Protection: 1; mode=block&quot;);<P>
X-Content-Type-Options:<BR>
Apache: Header always set X-Content-Type-Options: nosniff<P>
HTTP Strict-Transport-Security:<BR>
Apache: Header always set Strict-Transport-Security &quot;max-age=31536000; includeSubDomains&quot;<BR>
Nginx: add_header Strict-Transport-Security max-age=31536000;<P>
<B>Note: Network devices that include a HTTP/HTTPS console for administrative/management purposes often do not include all/some of the security headers. This is a known issue and it is recommend to contact the vendor for a solution. </B><P>]]></SOLUTION>
        <PCI_FLAG>1</PCI_FLAG>
        <LAST_UPDATE>2021-01-30T04:33:17Z</LAST_UPDATE>
        <CVSS_SCORE>
          <CVSS_BASE source="service">4.3 (AV:N/AC:M/Au:N/C:N/I:P/A:N)</CVSS_BASE>
          <CVSS_TEMPORAL>3.5 (E:U/RL:U/RC:UR)</CVSS_TEMPORAL>
        </CVSS_SCORE>
        <CVSS3_SCORE>
          **<CVSS3_BASE>5.3 (AV:N/AC:L/PR:N/UI:N/S:U/C:N/I:L/A:N)</CVSS3_BASE>**
          <CVSS3_TEMPORAL>4.7 (E:U/RL:U/RC:R)</CVSS3_TEMPORAL>
        </CVSS3_SCORE>
      </VULN_DETAILS>
```

**Steps to reproduce**

**Expected behavior**

**Deployment method** *(select with an `X`)*
- [ ] Kubernetes
- [ X] Docker
- [ ] setup.bash / legacy-setup.bash
I am using docker-compose.  
**Environment information**
 - Operating System: [e.g. Ubuntu 18.04]
 - DefectDojo Commit Message: [use `git show -s --format="[%ci] %h: %s [%d]"`]

**Sample scan files** (optional)
If applicable, add sample scan files to help reproduce your problem.

**Screenshots** (optional)

**Console logs** (optional)
If applicable, add console logs to help explain your problem.

**Additional context** (optional)

**Comment by alles-klar** [comment]
Qualys changed the CVSS_BASE field from integer to string in the last weeks...

**Comment by damiencarol** [comment]
@kareem-DA could you be more accurate on on the parser you used?
Also please fill completely the sections of the issue.

**Comment by kareem-DA** [comment]
> @kareem-DA could you be more accurate on on the parser you used?
> Also please fill completely the sections of the issue.

I updated the description.

**Comment by alles-klar** [comment]
@kareem-DA Can you share one of these "new" Qualys reports without confidential data here: https://github.com/DefectDojo/sample-scan-files/tree/master/qualys ? Then we can extend our tests and improve the parser.

**Comment by kareem-DA** [comment]
I have spent the better part of the day trying to pull our internal data out of the report and I haven't been able to without being sure that the report stays consistent.  Sorry

**Comment by alles-klar** [comment]
resolved in #4373
